airflow/configuration.py

```diff
@@ -442,20 +442,15 @@ def mkdir_p(path):
 
 
 # Setting AIRFLOW_HOME and AIRFLOW_CONFIG from environment variables, using
-# "~/airflow" and "~/airflow/airflow.cfg" respectively as defaults.
+# "~/airflow" and "$AIRFLOW_HOME/airflow.cfg" respectively as defaults.
 
-if 'AIRFLOW_HOME' not in os.environ:
-    AIRFLOW_HOME = expand_env_var('~/airflow')
-else:
-    AIRFLOW_HOME = expand_env_var(os.environ['AIRFLOW_HOME'])
+AIRFLOW_HOME = expand_env_var(os.environ.get('AIRFLOW_HOME', '~/airflow'))
 
 mkdir_p(AIRFLOW_HOME)
 
+AIRFLOW_CONFIG = expand_env_var(os.environ)
```

Comment:
This looks suspect - we're passing a dict to `expand_env_var()`, not a single value.